File: output/golang/struct/DeviceInformation.go
package main

type DeviceInformation struct {
	Platform_type string `json:"platform_type" bson:"platform_type"`
	Os_version string `json:"os_version" bson:"os_version"`
	Os_name string `json:"os_name" bson:"os_name"`
	Info_map map[string]string `json:"info_map" bson:"info_map"`
}

File: output/kotlin/models/overridden_serializable/DeviceInformation.kt
package overridden_serializable

interface IDeviceInformation {
    val platformType: String
    val osVersion: String?
    val osName: String
    val infoMap: Map<String, String>
}

@Serializable
data class DeviceInformation(
    @SerialName("platform_type") override val platformType: String,
    @SerialName("os_version") override val osVersion: String?,
    @SerialName("os_name") override val osName: String,
    @SerialName("info_map") override val infoMap: Map<String, String>,
) : IDeviceInformation